.github/workflows/run-benchmarks.yaml

```
@@ -0,0 +1,244 @@
+          USER_REPO_SHA: ${{ github.sha }}
+          WORKFLOW_REF: ${{ github.workflow_ref }}
+          ML_ACTIONS_REPO: google-ml-infra/actions
+        id: extract_ml_actions_repo_ref
+        shell: bash
+        run: |
+          set -euo pipefail
+          REPO_REF=""
+
+          if [[ "$USER_REPO" == "$ML_ACTIONS_REPO" ]]; then
+            # Use SHA for ML Actions PRs for reproducibility.
+            REPO_REF="$USER_REPO_SHA"
+          else
+            # For external repos, use ref (e.g. @v1).
+            REPO_REF=$(echo "$WORKFLOW_REF" | cut -d'@' -f2)
+          fi
+          
+          echo "repo_ref=$REPO_REF" >> "$GITHUB_OUTPUT"
+
+      - name: Checkout ML actions repo
+        uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # ratchet:actions/checkout@v5
+        with:
+          repository: 'google-ml-infra/actions'
+          ref: ${{ steps.extract_ml_actions_repo_ref.outputs.repo_ref}}
+          path: ml_actions
+          persist-credentials: false
+
+      - name: Generate benchmark matrix
+        env: 
+          REGISTRY_FILE: ${{ inputs.registry_file }}
+          WORKFLOW_TYPE: ${{ inputs.workflow_type }}
+        id: generate
+        shell: bash
+        run: |
+          set -euo pipefail
+          ML_ACTIONS_REPO="$GITHUB_WORKSPACE/ml_actions"
+          REGISTRY_PATH="$GITHUB_WORKSPACE/user_repo/$REGISTRY_FILE"
+          cd "$ML_ACTIONS_REPO" || exit 1
+
+          MATRIX_JSON="$(bazel run //benchmarking/gh_matrix_generator -- \
+          --registry_file="$REGISTRY_PATH" \
+          --workflow_type="$WORKFLOW_TYPE" | jq -c '.')"
+
+          echo "matrix=$MATRIX_JSON" >> "$GITHUB_OUTPUT"
+
+  run_benchmarks:
+    name: Run benchmark (${{ matrix.config_id }})
+    needs: generate_matrix
+    if: needs.generate_matrix.outputs.matrix != '[]'
+    strategy:
+      fail-fast: false
+      matrix:
+        include: ${{ fromJson(needs.generate_matrix.outputs.matrix) }}
+
+    runs-on: ${{ matrix.runner_label }}
```

Can you verify this works as intended if the caller is outside of this org? 

https://docs.github.com/en/actions/reference/workflows-and-actions/reusing-workflow-configurations#self-hosted-runners

---

Great question. I've verified our design already handles this.

The generate_matrix job (which runs on ubuntu-latest) acts as our gatekeeper. It uses the `GITHUB_REPOSITORY` variable to look up the calling repo (e.g. `openxla/tokamax`) in our gha_runners.json.

If the repo isn't in that config file (e.g., an external fork), the script fails, and the generate_matrix job fails. This prevents the run_benchmarks jobs from ever being queued.

So, it already fails fast for any unknown repository.